programming-mission/pr-review.md

```diff
@@ -0,0 +1,127 @@
+- Are all files opened and read outside of the handle function?
+- Does a CPAN module exist that we can use?
+    - Instead of implementing something yourself, check if a module already exists on CPAN
+- The Perl should not make any network requests
+    - We cannot allow the Perl to do this
+    - Any CPAN modules that make web requests are not allowed
+
+#### JavaScript:
+- Is the code wrapped in closure?
+- Is `use strict;` included at the top
+- Does this rely on any third party libraries?
+    - The developer should ask us before using external libraries. We need to add it internally if we agree to use it
+- Private variables and functions should be defined outside callback function
+- `Spice.failed()` should always be called when stopping execution prematurely
+- If the code needs to exit (e.g. the API response is missing required information) it must call `Spice.failed()`, so other Instant Answers are able to display
+- Experimental and browser-specific methods are not allowed. We need to ensure a consistent experience across all devices and browsers
+    - We support IE 9+, Edge, Chrome, Firefox, Safari, Opera, etc.
+- Will any code break if the API response is undefined?
+    - Make sure objects exist before accessing nested properties
+    - You can use `DDG.getProperty()` to get properties which may undefined (e.g. `DDG.getProperty(my_obj, 'some.property.name')`)
+
+#### CSS:
+- Is there any custom CSS?
+    - Custom CSS should be kept to a minimum. We prefer none, if possible
+- Are all CSS rules name-spaced?
+    - None of the CSS should be able to affect other Instant Answers or page elements
+- Font-colour, font-size and font-family should be defined by template variants, or our DDG classes (e.g. `.tx--14`, `.bg-clr--grey`, `.tx-clr--green`)
+- Template padding and margins should not be modified
+- Experimental and browser-specific methods are not allowed. We need to ensure a consistent experience across all devices & platforms
+    - We support IE 9+, Edge, Chrome, Firefox, Safari, Opera, etc.
+
+#### Tests:
+- Is a test file included?
+- Are all example queries covered by tests?
+- Are there tests for queries that **shouldn't** trigger the IA?
+    - Make sure edge cases and false positives are handled correctly
+
+#### Triggers:
+- List of dictionary words as triggers should be avoided (i.e. generic or ambiguous words that don't indicate search intent)
+- Use word/phrase triggers unless regex is absolutely neceseary
+    - Regexes should only be used when matching specific patterns or types of characters (numbers, unicode symbols)
+- Trigger words must be lowercase
+
+#### Cheat Sheets
+- Make sure the description is not redundant
+- Is the spelling correct?
+- "Title Case" should be used for name and sections
+- "Sentence case" should be used for description and explanations
+- Ensure the `CheatSheet.t` test passes
+- Use JSONLint to check the file and the text format
```

Comment:
@MariagraziaAlastra @moollaza [This](https://duckduckgo.com/?q=json+validate&t=h_&ia=answer) instead of some random JSONLint? 😉